src/vs/workbench/services/lifecycle/common/lifecycle.ts

```diff
@@ -65,9 +65,31 @@ export interface BeforeShutdownErrorEvent {
 	readonly error: Error;
 }
 
+export enum WillShutdownJoinerOrder {
+
+	/**
+	 * Joiners to run before the `Last` joiners.
+	 */
+	Default = 1,
+
+	/**
+	 * The joiners to run last.
```

Comment:
It might be useful to add a comment that the workbench will be disconnecting when this happens. A new consumer of joiners doesn't have any other guidance which is the right one to use for their case, and accidentally using a joiner that requires the workspace in `Last` would be easy to miss in testing.